org.uqbar.project.wollok/src/org/uqbar/project/wollok/WollokDsl.xtext

```diff
@@ -324,13 +330,25 @@ WCanExtendClass:
 // CLASSES
 
 WClass:
-	'class' name=ID ('inherits' parent=[WClass|QualifiedName])? '{'
+	'class' name=ID 
+		('inherits' parent=[WClass|QualifiedName])? 
+		('mixed with' mixins+=[WMixin|QualifiedName] (('and'|',') mixins+=[WMixin|QualifiedName])*)? 
+	'{'
 		(members+=WVariableDeclaration ';'?)*
 		(constructors+=WConstructor)*
 		(members+=WMethodDeclaration ';'?)*
 	'}'
 ;
 
+// MIXINS
+
+WMixin:
+	'mixin' name=ID '{'
```

Comment:
Ok, this one is heavy.
I agree that mixin is a general keyword that doesn't say anything.
On the other hand it is the most used term in programming literature.

I cannot think of a better term. I'm afraid that **behaviour** will collide with the speech when we say that an object has behaviour besides data.

As a **mixin** can model anything, it will be hard to find a good name.
Lets think. It doesn't depend on its content.
It could have more than one method, so it could add more than one behaviour to an object.
Maybe it is like a **feature** or **characteristic**, 
Sometimes it is used to model a **role**, although it seems to be a name which doesn't apply to all cases.
The main distinction from a class is that it cannot be instantiated, so it should model **part** of the behaviour of an object.
Maybe it is a **partial** ?